drgn/helpers/linux/cpumask.py

```diff
@@ -60,3 +60,50 @@ def for_each_possible_cpu(prog: Program) -> Iterator[int]:
 def for_each_present_cpu(prog: Program) -> Iterator[int]:
     """Iterate over all present CPUs."""
     return _for_each_cpu_mask(prog, "__cpu_present_mask")
+
+
+def cpumask_to_cpulist(cpumask: Object) -> str:
+    """
+    Get list of CPUs, present in a cpumask.
+
+    :param cpumask: ``struct cpumask*``
+    :returns: list of CPUs as string
+    """
+    start = 0
+    end = 0
+    count = 1
+    cpu_range = str()
+    all_cpu_ranges = str()
+    cpulist = [cpu for cpu in for_each_cpu(cpumask)]
+
+    if len(cpulist) == 1:  # Just one CPU in mask
+        return str(cpulist[0])
+
+    for index, value in enumerate(cpulist):
+        if index < len(cpulist) - 1:
+            if cpulist[index + 1] > value + 1:
+                end = index
+                if count > 1:
+                    cpu_range = str(cpulist[start]) + "-" + str(cpulist[end])
+                else:
+                    cpu_range = str(cpulist[start])
+                start = end + 1
+                count = 1
+                if not len(all_cpu_ranges):
+                    all_cpu_ranges += cpu_range
+                else:
+                    all_cpu_ranges += ", " + cpu_range
```

Comment:
CPU lists from the kernel don't usually have spaces, right?

I had an implementation for something similar lying around which might be a bit simpler:

```python3
def range_list_to_str(bits):
    start = end = -2
    parts = []
    for bit in bits:
        if bit == end + 1:
            end = bit
        else:
            if start >= 0:
                parts.append(str(start) if start == end else f"{start}-{end}")
            start = end = bit
    if start >= 0:
        parts.append(str(start) if start == end else f"{start}-{end}")
    return ",".join(parts)
```

Feel free to adapt that for this helper.

Reply:
Yes. CPU lists from kernel don't have spaces. I had added space as I thought it would be better ti show lists that way. But I agree with your point that we should as identical as possible to corresponding kernel interfaces. Have modified the helper as per your suggested logic above, so space is gone now :).